python/tank/authentication/user_impl.py

```diff
@@ -53,11 +53,11 @@ def __init__(self, host, http_proxy):
         # we don't end up infecting API instances with unicode strings
         # that would then cause some string data to be unicoded during
         # concatenation operations.
-        if isinstance(http_proxy, unicode):
-            http_proxy = http_proxy.encode("utf-8")
+        if http_proxy is not None:
+            http_proxy = six.ensure_str(http_proxy)
 
-        if isinstance(host, unicode):
-            host = host.encode("utf-8")
+        if host is not None:
```

Comment:
Some comment here

Reply:
In this case I think we can actually just omit the test, since there's this a few lines up:

```
if not host:
    raise IncompleteCredentials("missing host")
```

So I'll do that.